### ISSUE TITLE
Problems using allele frequency matrix as input

### ISSUE THREAD

**Issue by mae47** [JimWhiting91/afvaper#1]
Hi, thanks for sharing this software! 
I have been having a go with using an allele frequency matrix as the input file instead of a vcf, and had some problems with generating the "AF_input" list and the "null_input" list with the calc_AF_vectors function.

If not running null permutations, as when generating "AF_input" (ie line 252 of calc_AF_vectors.r script), line 261 throws up an error as R cannot compute subtraction with factors. If I am running null permutations, as when generating "null_input" (ie line 290), all the elements of the "null_input" list contain only zeros.

The following fix seems to work, but just wondering if I may have missed something or used a wrongly formatted input file?
[allele_frequency_matrix.txt](https://github.com/JimWhiting91/afvaper/files/9243839/allele_frequency_matrix.txt)

I edited the script to change the factors to numbers in line 261 (for AF_input):
`full_vectors[m,] <- AF_mat[,as.character(vectors[[m]][2])] - AF_mat[,as.character(vectors[[m]][1])]`
##changed to
`full_vectors[m,] <- as.numeric(as.character(AF_mat[,as.character(vectors[[m]][2])])) - as.numeric(as.character(AF_mat[,as.character(vectors[[m]][1])]))`

and the same for line 310 (for null_input):
`null_vectors[m,] <- null_AFs[,as.character(vectors[[m]][2])] - null_AFs[,as.character(vectors[[m]][1])]`
##changed to
`null_vectors[m,] <- as.numeric(as.character(null_AFs[,as.character(vectors[[m]][2])])) - as.numeric(as.character(null_AFs[,as.character(vectors[[m]][1])]))`

Thanks in advance,
Maddie

**Comment by JimWhiting91** [comment]
Hi Maddie,

Thanks for sending this through, and sorry for the delay in getting back to you, I've been away on holiday.

I've looked into this and there's a couple of things worth flagging. So the initial problem is being caused because your input file does have some issues where a number of lines are malformed, e.g. line 2550:
```
     CHROM     POS  pop0       pop1       pop2 pop3       pop4       pop5       pop6
2550    19 3636188 0,0.5 0.333333,0 0.166667,0  0,0 0.166667,0 0.166667,0 0.166667,0
```
Because in these lines, each individual frequency is recorded as `0,0.5`, these whole columns are being read in as character columns where they should be numeric. This then throws the error you're finding where those columns then can't be subtracted because they're the wrong class. Your approach does fix this, but it's my preference that something like this isn't implemented within the function because if the issue is that the input is malformed then I would rather it error out and users check their inputs, rather than force it to run and users not realise their input is malformed.

Another minor issues that I flagged is that I had intended the freq input files to have the same column names as in the manual, i.e. cols 1 and 2 named `chr` and `pos`, so this was also throwing up errors with your input. But to be honest, so long as the first column and second column contain chromosome and position info then the colnames shouldn't matter. I've updated this in the function and I'll release a minor update to reflect this.

Thanks again for flagging the issue,
Jim
